pkg/controller/plan/kubevirt.go

```diff
@@ -219,6 +219,37 @@ func (r *KubeVirt) GetImporterPod(pvc core.PersistentVolumeClaim) (pod *core.Pod
 	return
 }
 
+// Get the importer pods for a PersistentVolumeClaim.
+func (r *KubeVirt) GetImporterPods(pvc core.PersistentVolumeClaim) (pods []core.Pod, found bool, err error) {
+	pods = []core.Pod{}
+	if pvc.Annotations[AnnImporterPodName] == "" {
+		return
+	}
+
+	podList := &core.PodList{}
+
+	err = r.Destination.Client.List(
+		context.TODO(),
+		podList,
+		&client.ListOptions{
+			Namespace:     r.Plan.Spec.TargetNamespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{"app": "containerized-data-importer"}),
+		},
+	)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	for _, pod := range podList.Items {
+		if strings.Contains(pod.Name, pvc.Annotations["vmID"]) {
```

Comment:
looking at an existing importer pod and [here](https://github.com/kubevirt/containerized-data-importer/blob/18e6607f06d177abcb6b5efccc5e96bf59fb7bda/pkg/controller/import-controller.go#L1046-L1050), probably not - so if we cannot extend the label selector above, can we use a field selector?

Reply:
having more thought - this case may be problematic:
2 plans: A and B
in plan A we have a vm called: "liran"
in plan B we have a vm called: "liranr"

we ran plan B and it failed
we ran plan A and it passed
it will remove also the importer pod of plan B.

maybe there is another option to know which importer pods we need. but i am unsure about it.

Reply:
> i don't think we can. because it won't be equal to the pod name, it just look if the `vmID` is part of it.

yeah, I meant a field selector that selects by name and uses the ID of the VM with wildcards

Reply:
AFAIK, regex isn't supported (unless you use tools like `jq`)

Reply:
This options may fit: 
Change the string we look for to `importer-<plan_name>-<vmID>` prefix. that should limit the importer pods to the specific plan and specific vm.

Reply:
@liranr23 did you try something like they do with the namespaces [here](https://github.com/kubernetes/kubernetes/issues/107053#issuecomment-1193617828)?

Reply:
yes, that doesn't work(at least using kubectl/oc). the issue he mentioned there is closed due to inactivity.

Reply:
ok, on the bright side - the issue was accepted so we may get a fix later on
so as a temporary solution it's ok, but how about proposing to CDI the propagate labels from the DV to the importer pods to make it easier for us? it's done in a similar way with the virt-launcher pods